File: Problem_3/problem3.c
#include <stdio.h>

// gcc problem3.c -o problem3

int main() {
    unsigned long number = 600851475143;
    int factor = 2;
    while (factor < number) {
        while (number%factor == 0){
                number /= factor;
        }
        factor++;
    }
    
    printf("%lu\n", number);
    return 0;
}

File: Problem_3/problem3.ex
defmodule Problem do
  def solve(n), do: _solve(n, 2)
  def _solve(n, n), do: n
  
  def _solve(n, factor) do
    case rem(n, factor) do
      0 -> _solve(div(n, factor), factor)
      _ -> _solve(n, factor + 1)
    end
  end
end

number = 600851475143

IO.puts Problem.solve(number)